bin/base.sh

```diff
@@ -1,6 +1,12 @@
-CH_BIN="$(cd "$(dirname "$0")" && pwd)"
+libexec_base=$(dirname "$0")
+if [ -e ${libexec_base}/version.sh ]; then
+    . ${libexec_base}/version.sh
```

Comment:
I'm comfortable just sourcing it, without the error checking, since it shouldn't fail unless the install is wrong (i.e., users shouldn't see this and can't fix the error anyway) and I believe the scripts are `set -e`.

Reply:
Indeed. This allows to simplify the code significantly.